### ISSUE TITLE
[Bug]: selinux - enabling open-iscsi throws access denied

### ISSUE THREAD

**Issue by Keanu73** [kube-hetzner/terraform-hcloud-kube-hetzner#671]
### Description

I've tried to bootstrap a cluster twice with the same config, each time returning the same `access is denied` error when iscsid is attempted to be enabled. I've found that it's because SELinux is blocking the enabling of the service - when I disable SELinux (`setenforce 0`), it enables just fine.

Looks like the offending line is here:
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/4bc77b09d64031b74718a403b51a51d8edf81133/modules/host/main.tf#L153

I'm not sure whether it would be best to just throw in a manual SELinux policy override or whether there's some other issue at play?

```
type=AVC msg=audit(1679685463.126:138): avc:  denied  { add_name } for  pid=1 comm="systemd" name="iscsid.service" scontext=system_u:system_r:init_t:s0 tcontext=system_u:object_r:unlabeled_t:s0 tclass=dir permissive=0
type=AVC msg=audit(1679686040.541:383): avc:  denied  { add_name } for  pid=1 comm="systemd" name="iscsid.service" scontext=system_u:system_r:init_t:s0 tcontext=system_u:object_r:unlabeled_t:s0 tclass=dir permissive=0
```

### Kube.tf file

```terraform
locals {
  hcloud_token = ""
}

module "kube-hetzner" {
  providers = {
    hcloud = hcloud
  }
  hcloud_token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token

  source = "kube-hetzner/kube-hetzner/hcloud"

  ssh_public_key = file("~/.ssh/id_ed25519.pub")

  ssh_private_key = file("~/.ssh/id_ed25519")

  network_region = "eu-central" 

  control_plane_nodepools = [
    {
      name        = "control-plane-fsn1",
      server_type = "cpx11",
      location    = "fsn1",
      labels      = [],
      taints      = [],
      count       = 1

    }
  ]

  agent_nodepools = [
    {
      name        = "agent-large",
      server_type = "cpx21",
      location    = "fsn1",
      labels      = [],
      taints      = [],
      count       = 1

    },
    {
      name        = "storage",
      server_type = "cpx21",
      location    = "fsn1",

      labels      = [
        "node.kubernetes.io/server-usage=storage"
      ],
      taints      = [],
      count       = 1

    },

  ]

  control_planes_custom_config = {
   etcd-expose-metrics = true,
   kube-controller-manager-arg = "bind-address=0.0.0.0",
   kube-proxy-arg ="metrics-bind-address=0.0.0.0",
   kube-scheduler-arg = "bind-address=0.0.0.0",
  }

  enable_wireguard = true

  load_balancer_type     = "lb11"
  load_balancer_location = "fsn1"

  base_domain = "base.domain"

  autoscaler_nodepools = [
    {
      name        = "autoscaled-gameservers"
      server_type = "cpx31" 
      location    = "fsn1"
      min_nodes   = 1
      max_nodes   = 5
    }
  ]

  enable_longhorn = true

  ingress_controller = "nginx"

  ingress_replica_count = 0

  automatically_upgrade_os = false

  cni_plugin = "cilium"

   cilium_values = <<EOT
ipam:
  mode: kubernetes
devices: "eth1"
k8s:
  requireIPv4PodCIDR: true
kubeProxyReplacement: strict
l7Proxy: false
encryption:
  enabled: true
  type: wireguard
  EOT

}

provider "hcloud" {
  token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token
}

output "kubeconfig" {
  value     = module.kube-hetzner.kubeconfig
  sensitive = true
}
```

### Logs

```
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server: Provisioning with 'remote-exec'...
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): Connecting to remote host via SSH...
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Host: ip
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   User: root
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Password: false
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Private key: true
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Certificate: false
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   SSH Agent: false
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Checking Host Key: false
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec):   Target Platform: unix
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): Connected!
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): ++ systemctl list-units --all -t service --full --no-legend iscsid.service
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): ++ sed 's/^\s*//g'
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): ++ cut -f1 '-d '
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): + [[ iscsid.service == iscsid.service ]]
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): + systemctl enable --now iscsid
module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server (remote-exec): Failed to enable unit: Access denied
╷
│ Error: remote-exec provisioner error
│
│   with module.kube-hetzner.module.agents["1-0-storage"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 148, in resource "hcloud_server" "server":
│  148:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_267419275.sh": Process exited with status 1
╵
╷
│ Error: remote-exec provisioner error
│
│   with module.kube-hetzner.module.agents["2-0-egress"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 148, in resource "hcloud_server" "server":
│  148:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_1652559014.sh": Process exited with status 1
╵
╷
│ Error: remote-exec provisioner error
│
│   with module.kube-hetzner.module.agents["0-0-agent-large"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 148, in resource "hcloud_server" "server":
│  148:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_1908262381.sh": Process exited with status 1
╵
╷
│ Error: remote-exec provisioner error
│
│   with module.kube-hetzner.module.control_planes["0-0-control-plane-fsn1"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 148, in resource "hcloud_server" "server":
│  148:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_407957609.sh": Process exited with status 1
```

### Platform

Linux (WSL2)

**Comment by ron137** [comment]
Same happens to me

**Comment by mysticaltech** [comment]
Thanks for the debug info @Keanu73, on it.

**Comment by mysticaltech** [comment]
Folks, fix released in v1.10.8 🚀
